### PR TITLE
Create foundation for a difficulty calculator

### PR DESCRIPTION
**Pull request by Ryooshuu**
> There is no difficulty calculator, because I've talked to absolutely nobody about the mathematics behind it, and the current lazer taiko difficulty calculator uses a combined skill tocalculate its star rating.
> 
> I am not doing that.

Created on behalf of @irisu01

### CODE REVIEW

File: Kumi.Game/Gameplay/Difficulty/DifficultyCalculator.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+/// of a chart, such as tapping, streaming, or aiming. The strain values of each skill
+/// are then combined together to obtain the final strain values of the chart.
+///
+/// Each skill then returns a final difficulty result, which acts as a final value to
+/// be added to the final difficulty of the chart.
+///
+/// This class is abstract, as it does not implement any skills. Instead, it is up to
+/// <see cref="WorkingDifficultyCalculator"/> to implement the skills, and then the
+/// final result is returned by <see cref="Calculate"/>.
+/// </summary>
```
Comment (Ryooshuu) [inline Kumi.Game/Gameplay/Difficulty/DifficultyCalculator.cs:27]: Ain't reading all dat

**Review by Ryooshuu — COMMENTED** [review]
Would be nice to have some tests- oh shit I can't approve

**Comment by furiner** [comment]
> Would be nice to have some tests- oh shit I can't approve

chrome_4ahUFdXf45

File: Kumi.Game/Gameplay/Difficulty/DifficultyCalculator.cs (hunk 1 above)
Comment (furiner) [reply Kumi.Game/Gameplay/Difficulty/DifficultyCalculator.cs:27]: read it nerd

Comment (Ryooshuu) [reply Kumi.Game/Gameplay/Difficulty/DifficultyCalculator.cs:27]: i skimmed over it is that enough

**Review by furiner — APPROVED** [review]

**Pull request merged**